File: src/main/java/com/dev/k/invoice/invoice/entity/Invoice.java
package com.dev.k.invoice.invoice.entity;

import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import com.dev.k.invoice.common.entity.BaseEntity;
import com.dev.k.invoice.invoice.constant.InvoiceStatus;

@Entity
@Table(name = "invoices")
public class Invoice extends BaseEntity {

    @Id
    @Column(name = "invoice_id", nullable = false)
    private UUID invoiceId;

    @Column(name = "invoice_no", nullable = false, unique = true, length = 50)
    private String invoiceNo;

    @Column(name = "customer_id", nullable = false)
    private UUID customerId;

    @Column(name = "title", nullable = false, length = 255)
    private String title;

    @Column(name = "amount", nullable = false)
    private Integer amount;

    @Column(name = "due_date", nullable = false)
    private LocalDate dueDate;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private InvoiceStatus status;

    @Column(name = "issued_at", nullable = false)
    private OffsetDateTime issuedAt;

    @Column(name = "paid_at")
    private OffsetDateTime paidAt;

    public UUID getInvoiceId() {
        return invoiceId;
    }

    public void setInvoiceId(UUID invoiceId) {
        this.invoiceId = invoiceId;
    }

    public String getInvoiceNo() {
        return invoiceNo;
    }

    public void setInvoiceNo(String invoiceNo) {
        this.invoiceNo = invoiceNo;
    }

    public UUID getCustomerId() {
        return customerId;
    }

    public void setCustomerId(UUID customerId) {
        this.customerId = customerId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Integer getAmount() {
        return amount;
    }

    public void setAmount(Integer amount) {
        this.amount = amount;
    }

    public LocalDate getDueDate() {
        return dueDate;
    }

    public void setDueDate(LocalDate dueDate) {
        this.dueDate = dueDate;
    }

    public InvoiceStatus getStatus() {
        return status;
    }

    public void setStatus(InvoiceStatus status) {
        this.status = status;
    }

    public OffsetDateTime getIssuedAt() {
        return issuedAt;
    }

    public void setIssuedAt(OffsetDateTime issuedAt) {
        this.issuedAt = issuedAt;
    }

    public OffsetDateTime getPaidAt() {
        return paidAt;
    }

    public void setPaidAt(OffsetDateTime paidAt) {
        this.paidAt = paidAt;
    }
}

File: frontend/src/types/invoice.ts
/**
 * 請求ステータスの型定義
 */
export type InvoiceStatus = 'ISSUED' | 'PARTIALLY_PAID' | 'PAID' | 'OVERDUE'

/**
 * 請求一覧・詳細で使用する請求情報の型定義
 */
export interface Invoice {
  invoiceId: string
  invoiceNumber: string
  customerId: string
  title: string
  amount: number
  paidAmount: number
  balance: number
  status: InvoiceStatus
  dueDate: string
  issuedAt: string
  paidAt?: string | null
}

/**
 * 請求一覧表示用のサマリー型
 *
 * バックエンドの InvoiceSummaryResponse に対応
 * 一覧画面で使用する軽量データ
 *
 * ※ 詳細画面用の Invoice とは別
 */
export type InvoiceSummary = {
  /** 請求ID（UUID） */
  invoiceId: string

  /** 請求番号 */
  invoiceNo: string

  /** 顧客ID（UUID） */
  customerId: string

  /** 請求タイトル */
  title: string

  /** 請求金額 */
  amount: number

  /** 支払期限（ISO文字列） */
  dueDate: string

  /** ステータス（未払い / 支払済 / 期限切れなど） */
  status: string
}

/**
 * 請求検索条件の型定義
 */
export interface InvoiceSearchParams {
  customerId?: string
  status?: InvoiceStatus | ''
  dueDateFrom?: string
  dueDateTo?: string
}

/**
 * 請求登録リクエストの型定義
 */
export interface CreateInvoiceRequest {
  invoiceNumber: string
  customerId: string
  title: string
  amount: number
  dueDate: string
}
